vdr/src/contracts/did/types/did_doc.rs

```diff
@@ -1,35 +1,92 @@
 
 pub const BASE_CONTEXT: &str = "https://www.w3.org/ns/did/v1";
 pub const SECPK_CONTEXT: &str = "https://w3id.org/security/suites/secp256k1recovery-2020/v2";
 pub const KEYS_CONTEXT: &str = "https://w3id.org/security/v3-unstable";
 
-#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct DidDocumentWithMeta {
-    pub did_document: DidDocument,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub did_document: Option<DidDocument>,
     pub did_document_metadata: DidMetadata,
     pub did_resolution_metadata: DidResolutionMetadata,
 }
 
-#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DidResolutionMetadata {
+    pub content_type: Option<DidResolutionAcceptType>,
+    pub error: Option<DidResolutionError>,
+    pub message: Option<String>,
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct DidResolutionOptions {
-    pub accept: String,
+    pub accept: DidResolutionAcceptType,
+    pub block_tag: Option<Block>,
 }
 
 #[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
-pub struct DidResolutionMetadata {
-    pub content_type: String,
-    pub error: Option<String>,
+pub enum DidResolutionAcceptType {
+    #[default]
+    #[serde(rename = "application/did+ld+json")]
+    DidLdJson,
+}
+
+impl TryFrom<&str> for DidResolutionAcceptType {
+    type Error = VdrError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "application/did+ld+json" => Ok(DidResolutionAcceptType::DidLdJson),
+            value => Err(VdrError::CommonInvalidData(format!(
+                "Unsupported `accept` value {} provided for DID resolution type.",
+                value
+            ))),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub enum DidResolutionError {
+    /*
+     * The resolver has failed to construct the DID document.
+     * This can be caused by a network issue, a wrong registry address or malformed logs while parsing the registry
+     * history. Please inspect the `DIDResolutionMetadata.message` to debug further.
+     */
+    #[serde(rename = "notFound")]
+    NotFound,
+    /*
+     * The resolver does not know how to resolve the given DID. Most likely it is not a `did:ethr`.
+     */
+    #[serde(rename = "invalidDid")]
+    InvalidDid,
+    /*
+     * The resolver does not support the 'accept' format requested with `DidResolutionOptions`.
+     */
+    #[serde(rename = "representationNotSupported")]
+    RepresentationNotSupported,
+    /*
+     * The resolver is misconfigured or is being asked to resolve a `DID` anchored on an unknown network
+     */
+    #[serde(rename = "unknownNetwork")]
+    UnknownNetwork,
```

Comment:
This error not from the spec https://www.w3.org/TR/did-spec-registries/#did-resolution-metadata
Don't we want to use `RepresentationNotSupported` instead of it?